### PR TITLE
Fix WorkspaceIndex not reindexing when document changes.

### PR DESCRIPTION
**Pull request by BladeMF**

### CODE REVIEW

File: lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php [hunk 1]
```diff
@@ -37,41 +46,63 @@ public function documentForName(Name $name): ?TextDocument
...
+            $this->byName[$name] = $textDocument;
+        }
+        foreach($namesToRemove as $name){
+            unset($this->byName[$name]);
+        }
+
+        if(!empty($newNames))
+            $this->documentToNameMap[(string)$textDocument->uri()] = $newNames;
+        else
+            unset($this->documentToNameMap[(string)$textDocument->uri()]);
```
Comment (dantleech) [inline lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php:80]: cs fixer is failing, probably this is one of them, just run `./vendor/bin/php-cs-fixer fix` (or `composer integrate`)

**Comment by dantleech** [comment]
I'll need to check to see the performance regression (if any) with this.

**Comment by BladeMF** [comment]
I wouldn't expect to have any regression in performance. We are not `foreach`-ing the index now, so it should be faster, but using a bit more memory.

File: lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php [hunk 2]
```diff
@@ -37,41 +45,64 @@ public function documentForName(Name $name): ?TextDocument
...
+        foreach ($newNames as $name) {
+            $this->byName[$name] = $textDocument;
+        }
+        foreach ($namesToRemove as $name) {
+            unset($this->byName[$name]);
+        }
+
+        if (!empty($newNames)) {
+            $this->documentToNameMap[(string)$textDocument->uri()] = $newNames;
+        } else {
```
Comment (dantleech) [inline lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php:79]: minor but I would remove else in favor of an early return:

```
if ($something) {
    // do something
    return;
}

// do something else
```

**Comment by dantleech** [comment]
Good job :+1:

File: lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php (hunk 2 above)
Comment (BladeMF) [reply lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php:79]: Sure, but I am not sure how to append this to the current PS - push to the master branch again?

Comment (dantleech) [reply lib/LanguageServerWorseReflection/Workspace/WorkspaceIndex.php:79]: It's fine, it's merged now.

**Pull request merged**